### PR TITLE
chore: optimize circle ci jobs to increase performance and cleanup incorrect configurations

### PR DESCRIPTION
**Pull request by brian-mann**
### Default Resource Class
- Set `resource_class` to be `medium` by default as most of our jobs ended up using `medium`, and some jobs that should have been `medium` were really `small`. The exception should be something other than `medium`.

### Increase Parallelism
- Increased parallelism on several jobs were long running jobs blocking others:
  - `run-launchpad-integration-tests-chrome`
  - `run-frontend-shared-component-tests-chrome` 
  - `run-app-integration-tests-chrome`
  - `reporter-integration-tests`

### Decrease Parallelism
- Removed parallelism of several jobs because they were not parallelizing anything and instead were duplicating the exact same steps N number of times, increasing resource cost for zero gain
  - `npm-vue`
  - `cli-visual-tests`

### Increase Resource Class
- Several jobs were previously on `small`, which I have bumped to `medium` because they are either running Cypress tests, were fairly long-running, were maxing out their CPU, or because they stand in the critical path blocking other jobs
  -  `ui-components-integration-tests` was running Cypress tests
  - `npm-vite-dev-server` was running Cypress tests
  - `npm-angular` was running Cypress tests and fairly long running
  - `check-ts` fairly long running, pegged CPU
  - `server-performance-tests` fairly long running, pegged CPU
  - `server-unit-tests` fairly long running, pegged CPU
  - `linux-lint` fairly long running, pegged CPU
- One job is in the critical path and blocks other jobs and its worthwhile to try to squeeze out all time savings we can at the expensive of higher resource cost
  -  `system-tests-node-modules-install` was bumped from `small` to `medium` 

### Decrease Resource Class
- Several jobs were downgraded to `small` from `medium` because they don't need extra resources
  - `unit-tests-release`

## Needs Additional Eyes

cc @lmiller1990 and @elevatebart 

I couldn't tell offhand, but from what I read in the code I believe that `npm-react` is not actually testing all of the examples we intended it to test. It uses `CIRCLE_NODE_INDEX` to manually "chunk" and parallelize the example tests for vue across around 8 machines, but I don't think the logic is written correctly, and in fact it's **only running** up to 8 examples and not all of them. I could be wrong though.

### CODE REVIEW

File: packages/app/cypress.config.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default defineConfig({
     runMode: 2,
     openMode: 0,
   },
+  videoCompression: false, // turn off video compression for CI
```
Comment (tgriesser) [inline packages/app/cypress.config.ts:13]: Is there a reason this isn't the default in Cypress?

**Comment by lmiller1990** [comment]
@brian-mann  I think you are correct, looking at CI, we only have 8 containers for `npm/react`. We should have... 9,  see `npm/react/examples.env` for what should run.

```
/examples/nextjs
/examples/react-scripts
/examples/react-scripts-typescript
/examples/find-webpack
/examples/webpack-file
/examples/react-scripts-folder
/examples/using-babel-typescript
/examples/webpack-options
/examples/sass-and-ts
```

Looks like we were missing sass-and-ts. As part of another ticket, UNIFY-1180, I'm moving some CT examples into `system-tests`, hopefully we can move everything there soon and not need to do the weird chunking we currently are doing for `npm/react`.

File: packages/app/cypress.config.ts (hunk 1 above)
Comment (brian-mann) [reply packages/app/cypress.config.ts:13]: historical reasons and people not recording to the dashboard

**Review by tgriesser — DISMISSED** [review]
Seems fine to me, @flotwig might be able to give better thoughts after looking into the optimization here for awhile.

File: circle.yml [hunk 2]
```diff
@@ -61,8 +61,8 @@ executors:
   cy-doc:
     docker:
       - image: cypress/browsers:node16.5.0-chrome94-ff93
-    # by default, we use "small" to save on CI costs. bump on a per-job basis if needed.
-    resource_class: small
+    # by default, we use "medium" to balance performance + CI costs. bump or reduce on a per-job basis if needed.
+    resource_class: medium
```
Comment (flotwig) [inline circle.yml:65]: This is also the CircleCI default, so these lines could be removed:

```suggestion
```

File: circle.yml [hunk 3]
```diff
@@ -1055,7 +1056,8 @@ jobs:
 
   cli-visual-tests:
     <<: *defaults
-    parallelism: 8
+    resource_class: small
+    parallelism: 1
```
Comment (flotwig) [inline circle.yml:1060]: This is the default, can remove this:

```suggestion
```

File: circle.yml [hunk 4]
```diff
@@ -1463,8 +1452,7 @@ jobs:
 
   npm-vue:
     <<: *defaults
-    resource_class: medium
-    parallelism: 3
+    parallelism: 1
```
Comment (flotwig) [inline circle.yml:1455]: ditto

```suggestion
```

File: circle.yml [hunk 5]
```diff
@@ -1198,37 +1198,34 @@ jobs:
 
   system-tests-node-modules-install:
     <<: *defaults
+    resource_class: medium+ ## trying this out to see if it makes a difference since this is a critical blocking job
```
Comment (flotwig) [inline circle.yml:1201]: Did this yield any positive results? Looking at the output it seems like it's not making full use of `medium+`: https://app.circleci.com/pipelines/github/cypress-io/cypress/32461/workflows/d2c7ea79-3f7e-4c0f-a50a-fa65b8853ce7/jobs/1276173/resources

File: circle.yml (hunk 4 above)
Comment (flotwig) [inline circle.yml:1455]:
```suggestion
```

**Review by flotwig — APPROVED** [review]

File: circle.yml (hunk 2 above)
Comment (brian-mann) [reply circle.yml:65]: i wanted the line / explanation in there for clarity

**Review by flotwig — APPROVED** [review]

**Review by tgriesser — APPROVED** [review]

**Pull request merged**